File: server/models/profile/googleUser.js
export default function buildMakeGoogleUser() {
  return function makeGoogleUser({
    username,
    email,
    photoUrl,
    googleId,
    displayName,
    postsRecordId,
    networkRepoId,
  }) {
    if (username == null) {
      throw new Error("username_empty");
    }

    if (username.length == 0) {
      throw new Error("username_empty");
    }

    if (username.length > 20) {
      throw new Error("username_outOfRange");
    }

    if (email == null || email.length == 0) {
      throw new Error("email_empty");
    }

    if (googleId == null || googleId.length == 0) {
      throw new Error("googleId_empty");
    }

    return Object({
      username,
      email,
      photoUrl,
      googleId,
      displayName,
      postsRecordId,
      networkRepoId,
    });
  };
}

File: lib/backend/Models/authentication/GoogleUser.dart
import './OAuthUser.dart';

class GoogleUser extends OAuthUser {
  String accessToken;
  String googleId;
  String photoUrl;
  String username;
  String email;
  String displayName;

  @override
  String toString() {
    return "{accessToken:$accessToken,googleId:$googleId,photoUrl:$photoUrl,username:$username,email:$email,displayName:$displayName}";
  }
}
